stdlib/mmap.pyi

```diff
@@ -27,7 +27,7 @@ if sys.platform != "win32":
 
     PAGESIZE: int
 
-class mmap(AbstractContextManager[mmap], Iterable[int], Sized):
+class mmap(AbstractContextManager[mmap], ReadableBuffer, Iterable[int], Sized):
```

Comment:
Another thought: how does this even work? `ReadableBuffer` is a `Union` which includes `mmap.mmap`, which leads me to the conclusion that this should result in some sort of infinite type loop.

Reply:
Update: this does lead to an error :) Will probably split up `WriteableBuffer` to fix this.

Reply:
We should add it to the union, not use the union as a base class.

Reply:
What do you mean? `mmap.mmap` is already part of `WriteableBuffer`.

Reply:
Oh, right. Then we shouldn't change it at all.

The original problem was that mmap wasn't accepted in re.match. If re.match accepts all buffers, then we should change its signature, not anything about mmap.

Reply:
Upon further research, it seems to me like a way to solve this would be to have `mmap.mmap` extend `ByteString`, and change the definition of `AnyStr` to:

```python
AnyStr = TypeVar("AnyStr", str, ByteString)
```

However, that would require a change in CPython. I could also create an `AnyStr` alternative, specific to `Pattern`, with this definition. One last option would be to have `mmap.mmap` extend `bytes` (aka `ReadOnlyBuffer`). What do you think?

Reply:
We should not change AnyStr. That would require CPython runtime changes. We could use a separate typevar specific to regexes, but I'd have to look more into it to figure out whether that's feasible.